keps/sig-cli/kustomize-default-config.md

```diff
@@ -0,0 +1,314 @@
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternatives](#alternatives)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+Kustomize builtin transformers are provided with default configurations describing which
+resource fields are subject of a particular transformation. kustomize provides the ability
+to add fields to the default configuration but does not let the user adjust that configuration
+in order to skip a transformation for a specific resource or resource kind.
+
+We are aiming at addressing those issues:
+- Make cluster level kind configurable [#617](https://github.com/kubernetes-sigs/kustomize/issues/617)
+- Do not add namespace to cluster scoped CRDs [#552](https://github.com/kubernetes-sigs/kustomize/issues/552) 
+- Skipping nameprefix for some resource kind [#519](https://github.com/kubernetes-sigs/kustomize/issues/617)
+- Adding labels only to metadata.labels [#330](https://github.com/kubernetes-sigs/kustomize/issues/330)
+- Skip adding labels to certain paths [#519](https://github.com/kubernetes-sigs/kustomize/issues/519)
+- Unable to disable commonLabels injection using transformer config: Error: conflicting fieldspecs [#817](https://github.com/kubernetes-sigs/kustomize/issues/817)
+
+## Motivation
+
+
+### Goals
+
+- Provide the ability for the user to skip transformation for specific group/version/kind especially usefull if the default
+  configuration is using a wild card (see namespace and prefixsuffix transformer configuration).
+- Provide the ability of the user to replace or remove an entry in the default configuration which is not matching his needs
+  (see commonLabel transformer configuration)
+
+  
+### Non-Goals
+
+- Provide backward compatibility with the transformer default configurations.
+- Provide a way for the user to exclude a kind from a wild card based set as opposed to have to list all the possible kinds/gvks.
+- Provide a way for the user to remove a configuration entry from the default configuration if it is not aligned with the needs
+  of his project.
+- Provide a simple method to the transformer developers to select the list of "mutable" fields based on the gvk of the resources.
+
+## Proposal
+
+The propsal has three parts:
+
+1. Update the FieldSpec definition to add a "skip" field as follow (It is also useful external transformers needing wild card):
+
+```go
+type FieldSpec struct {
+	gvk.Gvk            `json:",inline,omitempty" yaml:",inline,omitempty"`
+	Path               string `json:"path,omitempty" yaml:"path,omitempty"`
+	CreateIfNotPresent bool   `json:"create,omitempty" yaml:"create,omitempty"`
```

Comment:
What's the relationship between `CreateIfNotPresent` and `SkipTransformation`? Which one has higher priority?